src/app/staking/[addressOrENSName]/page.jsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import Tenant from "@/lib/tenant/tenant";
+import { apiFetchStakedDeposits } from "@/app/api/staking/getStakedDeposits";
+import { fetchDelegate } from "@/app/api/common/delegates/getDelegates";
+import { HStack } from "@/components/Layout/Stack";
+import { DepositList } from "@/app/staking/[addressOrENSName]/deposits/DepositList";
+import { StakingStats } from "@/app/staking/components/StakingStats";
+import StakingFaq from "@/app/staking/components/StakingFaq";
+import { PanelClaimRewards } from "@/app/staking/components/PanelClaimRewards";
+import { PanelNewDeposit } from "@/app/staking/components/PanelNewDeposit";
+import { resolveENSName } from "@/app/lib/ENSUtils";
+import { revalidatePath } from "next/cache";
+import { StakingIntro } from "@/app/staking/components/StakingIntro";
+import { PanelSetAllowance } from "@/app/staking/components/PanelSetAllowance";
+
+async function fetchDeposits(address) {
+  "use server";
+  return apiFetchStakedDeposits({ address });
+}
+
+async function apiFetchDelegate(address) {
+  "use server";
+  return fetchDelegate(address);
+}
+
+export default async function Page({ params: { addressOrENSName } }) {
+  const address = (await resolveENSName(addressOrENSName)) || addressOrENSName;
+
+  const { token } = Tenant.current();
+  const { ui, contracts } = Tenant.current();
+  const deposits = await fetchDeposits(address.toLowerCase());
```

Comment:
could probably put this async action below the check for staking toggle but its not a huge deal -- might just make slightly better perf to get hit with a "route not supported" lol